install/cupy_builder/build_amd_utils.py

```diff
@@ -0,0 +1,30 @@
+import os
+
+hipified_sources = [
+    "cupy_backends/cuda/libs/curand_hip.pyx"
+]
+
+def modify_hip_gen_code(path):
```

Comment:
We might still need this logic as once we generate hipified pyx/pxd files, since CUPY_INSTALL_USE_HIP is on it would try to import itself which would lead to circular imports. So we might need this switch.